modules/benchmark/R/calc.benchmark.R

```diff
@@ -3,53 +3,65 @@
 ##'  
 ##' @name calc.benchmark 
 ##' @title Calculate benchmarking statistics
-##' @param settings object from start.benchmark.runs (for now - we many only need a few variables)
+##' @param bm.ensemble object, either from create.BRR or start.bm.ensemle
 ##' @param con database connection
 ##' @export 
 ##' 
 ##' @author Betsy Cowdery 
 
-calc.benchmark <- function(settings, con){ #settings file is output from start.benchmark.runs
+calc.benchmark <- function(bm.ensemble, con){ 
   
   require(RPostgreSQL)
   require(XML)
   require(data.table)
   
   
-  # Query BRR for benchmark ids
-  
-  bm.id <- settings$benchmark$id 
-  obvs.id <- settings$benchmark$input_id
-  
-  
-  # For each bm.id - don't have this written as a loop yet. 
-  #  Query database for: input path, variables, site, start/end dates, metrics, formats
-  
-  data.path <- query.file.path(obvs.id,settings$host$name,con)
-  format <- query.format.vars(obvs.id,con)  
-  
-  site  <- query.site(settings$run$site$id, con)
-  
-  metrics <- db.query(paste("SELECT m.name from metrics as m JOIN benchmarks_metrics as b 
-                              ON m.id = b.metric_id WHERE b.benchmark_id = ", bm.id),con)
-  
-  model_run <- dir(settings$host$outdir, full.names=TRUE)[1]
+  ens <- db.query(paste("SELECT * FROM ensembles where id = ",bm.ensemble$ensemble_id,";"), con) 
+  wf <- db.query(paste("SELECT w.* FROM workflows as w join ensembles as e on w.id = e.workflow_id where e.id = ",bm.ensemble$ensemble_id,";"), con)
+  site  <- query.site(wf$site_id, con)
+  start_year <- year(wf$start_date)
+  end_year <- year(wf$end_date)
+  model_run <- dir(file.path(wf$folder,"out"), full.names=TRUE,  include.dirs = TRUE)[1]
   #How are we dealing with ensemble runs? Right now I've hardcoded to select the first run.
   
-  # Local or remote
-  # format is a list, metrics is a dataframe, both can be long. How to pass to calc.metrics remotely? 
-  # Maybe as an Rdata file?
+  # All benchmarking records for the given benchmarking ensemble id
+  bms <- db.query(paste("SELECT * FROM benchmarks as b join benchmarks_benchmarks_reference_runs as r on b.id=r.benchmark_id  join  benchmarks_ensembles as be on r.reference_run_id = be.reference_run_id where be.ensemble_id = ",bm.ensemble$ensemble_id,";"), con) 
+  
+  
+  # Determine how many data sets inputs are associated with the benchmark id's
+  inputs <- unique(bms$input_id)
+  # bm.ids are split up in to groups according to their input data. 
+  # So that the input data is only loaded once. 
```

Comment:
I understand this logic, but when I look below, it appears that even though you've lumped the analysis by input, you are not loading the data before looping over the individual inputs, but rather just looking up the path to the inputs once and then loading it numerous times.